projects/igniteui-angular/src/lib/input-group/input-group.component.ts

```diff
@@ -174,12 +207,23 @@ export class IgxInputGroupComponent extends DisplayDensityBase implements IgxInp
      * @hidden
      */
     @HostListener('click', ['$event'])
-    public onClick(event) {
-        if (!this._supressInputAutofocus) {
+    public onClick(event: MouseEvent) {
+        // if the click target is not the input, e.g. click on prefix, focus the input.
+        if (event.target !== this.input.nativeElement || !this._supressInputAutofocus) {
```

Comment:
This isn't compatible with the current `onPointerDown` logic that will always prevent on prefix/suffix/label clicks as this will not trigger and when it does - it's on the input and it seems like it'll do nothing

Reply:
I was going to suggest dropping the entire `supressInputAutofocus` functionality, but I came up with at least one good reason to keep it - namely to maintain an option to use buttons/icons in the input group that can perform action without focusing the input. That's particularly useful for mobile devices to prevent the software keyboard popping up - similar to what we've done for the combo search and before.

@wnvko Ignore the deprecation note for the changelog and revert it instead, logic on the handlers should be updated. I'd make the pointer prevent only active when already focused.

Reply:
done